core/src/main/java/kafka/automq/table/process/transform/DebeziumUnwrapTransform.java

```diff
@@ -0,0 +1,297 @@
+                if (beforeDelete == null) {
+                    throw new TransformException("Invalid DELETE record: missing required 'before' data");
+                }
+                return beforeDelete;
+
+            default:
+                log.warn("Unknown Debezium operation type: {}. Attempting to use 'after' data", operation);
+                GenericRecord fallback = getRecordValue(record, FIELD_AFTER);
+                if (fallback == null) {
+                    throw new TransformException("Unsupported operation '" + operation + "' with no usable data");
+                }
+                return fallback;
+        }
+    }
+
+
+    private GenericRecord enrichWithMetadata(GenericRecord businessData,
+                                           GenericRecord debeziumRecord,
+                                           String operation,
+                                           TransformContext context) throws TransformException {
+        try {
+            Schema schemaWithMetadata = createSchemaWithMetadata(businessData.getSchema());
+            GenericRecordBuilder builder = new GenericRecordBuilder(schemaWithMetadata);
+
+            for (Schema.Field field : businessData.getSchema().getFields()) {
+                builder.set(field.name(), businessData.get(field.name()));
+            }
+
+            Schema cdcSchema = schemaWithMetadata.getField(CDC_RECORD_NAME).schema();
+            GenericRecordBuilder cdcBuilder = new GenericRecordBuilder(cdcSchema);
+
+            cdcBuilder.set(CDC_FIELD_OP, mapOperation(operation));
+
+            Object tsMs = debeziumRecord.get(FIELD_TS_MS);
+            if (tsMs instanceof Long) {
+                cdcBuilder.set(CDC_FIELD_TS, tsMs);
+            }
+
+            cdcBuilder.set(CDC_FIELD_OFFSET, context.getKafkaRecord().offset());
+
+            GenericRecord source = getRecordValue(debeziumRecord, FIELD_SOURCE);
+            if (source != null) {
+                String schema = getStringValue(source, "schema");
+                String db = (schema == null) ? getStringValue(source, "db") : schema;
+                String table = getStringValue(source, "table");
+                if (db != null && table != null) {
+                    cdcBuilder.set(CDC_FIELD_SOURCE, db + "." + table);
+                }
+            }
+
+            builder.set(CDC_RECORD_NAME, cdcBuilder.build());
+            return builder.build();
+
+        } catch (Exception e) {
+            throw new TransformException("Failed to enrich record with Debezium metadata", e);
+        }
+    }
+
+    private String mapOperation(String originalOp) {
+        switch (originalOp) {
+            case "u":
+                return "U";
+            case "d":
+                return "D";
+            default:
+                // Debezium ops "c", "r", and any others
+                return "I";
+        }
+    }
+
+    private Schema createSchemaWithMetadata(Schema originalSchema) {
```

Comment:
Maybe we could enhance the Schema & GenericRecord to avoid copy the fields in future.